src/code/code_800EC960.c

```diff
@@ -1203,916 +1233,1080 @@ u8 D_8016B9F4;
+        if ((sCurOcarinaPitch != OCARINA_PITCH_NONE) && CHECK_BTN_ANY(sOcarinaInputButtonCur, BTN_R) &&
+            (sRecordingState != OCARINA_RECORD_SCARECROW_SPAWN)) {
+            sCurOcarinaButtonIndex += 0x80; // Flag to resolve B Flat 4
+            sCurOcarinaPitch++;             // Raise the pitch by 1 semitone
+        }
+
+        // Pressing the Z Button will lower the pitch by 1 semitone
+        if ((sCurOcarinaPitch != OCARINA_PITCH_NONE) && CHECK_BTN_ANY(sOcarinaInputButtonCur, BTN_Z) &&
+            (sRecordingState != OCARINA_RECORD_SCARECROW_SPAWN)) {
+            sCurOcarinaButtonIndex += 0x40; // Flag to resolve B Flat 4
+            sCurOcarinaPitch--;             // Lower the pitch by 1 semitone
+        }
+
+        if (sRecordingState != OCARINA_RECORD_SCARECROW_SPAWN) {
+            // Bend the pitch of the note based on y control stick
+            sCurOcarinaBendIndex = sOcarinaInputStickRel.y;
+            sCurOcarinaBendFreq = AudioOcarina_BendPitchTwoSemitones(sCurOcarinaBendIndex);
+
+            // Add vibrato of the ocarina note based on the x control stick
+            sCurOcarinaVibrato = ABS_ALT(sOcarinaInputStickRel.x) >> 2;
+            // Sets vibrato to io port 6
+            Audio_QueueCmdS8(0x6 << 24 | SEQ_PLAYER_SFX << 16 | SFX_CHANNEL_OCARINA << 8 | 6, sCurOcarinaVibrato);
         } else {
-            D_80130F2C = 0;
-            D_80130F24 = 1.0f;
+            // no bending or vibrato for recording state OCARINA_RECORD_SCARECROW_SPAWN
+            sCurOcarinaBendIndex = 0;
+            sCurOcarinaBendFreq = 1.0f; // No bend
         }
 
-        if ((sCurOcarinaBtnVal != 0xFF) && (sPrevOcarinaNoteVal != sCurOcarinaBtnVal)) {
-            Audio_QueueCmdS8(0x6 << 24 | SEQ_PLAYER_SFX << 16 | 0xD07, D_80130F10 - 1);
-            Audio_QueueCmdS8(0x6 << 24 | SEQ_PLAYER_SFX << 16 | 0xD05, sCurOcarinaBtnVal);
-            Audio_PlaySoundGeneral(NA_SE_OC_OCARINA, &D_801333D4, 4, &D_80130F24, &D_80130F28, &D_801333E8);
-        } else if ((sPrevOcarinaNoteVal != 0xFF) && (sCurOcarinaBtnVal == 0xFF)) {
+        // Processes new and valid notes
+        if ((sCurOcarinaPitch != OCARINA_PITCH_NONE) && (sPrevOcarinaPitch != sCurOcarinaPitch)) {
+            // Sets ocarina instrument Id to channelIdx io port 7, which is used
+            // as an index in seq 0 to get the true instrument Id
+            Audio_QueueCmdS8(0x6 << 24 | SEQ_PLAYER_SFX << 16 | SFX_CHANNEL_OCARINA << 8 | 7, sOcarinaInstrumentId - 1);
+            // Sets pitch to io port 5
+            Audio_QueueCmdS8(0x6 << 24 | SEQ_PLAYER_SFX << 16 | SFX_CHANNEL_OCARINA << 8 | 5, sCurOcarinaPitch);
+            Audio_PlaySoundGeneral(NA_SE_OC_OCARINA, &D_801333D4, 4, &sCurOcarinaBendFreq, &sRelativeOcarinaVolume,
+                                   &D_801333E8);
+        } else if ((sPrevOcarinaPitch != OCARINA_PITCH_NONE) && (sCurOcarinaPitch == OCARINA_PITCH_NONE)) {
+            // cancels ocarina for a non-valid input
```

Comment:
afaict this also happens when the player stops pressing anything on the controller?

Reply:
Correct, which will set `sCurOcarinaPitch = OCARINA_PITCH_NONE`, I can clarify